### PR TITLE
Move JourneyMap class reference out of code that runs even if it's not installed

### PR DESCRIPTION
**Pull request by ByThePowerOfScience**
Currently, GTNH always ships with this mod installed, but it doesn't _necessarily_ include JourneyMap.  Some launchers endorse a different mapping mod and remove JourneyMap automatically, and some players just remove it on their own.  

Due to functions that get run without having JourneyMap installed including a JourneyMap class import, clients without JourneyMap get repeated `ClassNotFoundException` crashes whenever a waypoint is shared in chat.

NOTE: For some reason, I wasn't able to download the POMs for the dependencies (even after several hours of waiting), so I wasn't able to actually get the build environment running on my computer meaning I basically wrote this in Notepad.  It is very likely that this code doesn't work, so **please make sure it compiles before merging!!!**

Changes:
- Removed all references to the `journeymap.client.model.Waypoint` class from `MessageShareWaypoint.kt`.
- Moved `Waypoint.fromString()` in `MessageShareWaypoint.kt` to an alias function `WaypointUtil.waypointFromString()`.
- Made `MessageShareWaypoint#handleClientSideMessage` only fire if JourneyMap is loaded, and cached the `Loader#isModLoaded` result in `WaypointUtil`.

### CODE REVIEW

**Comment by chochem** [comment]
> Currently, GTNH always ships with this mod installed, but it doesn't necessarily include JourneyMap

GTNH always ships with journeymap. no plans to change that either.

**Comment by mitchej123** [comment]
If I had to guess it's ic2's maven.  It'll work if you updated the bulldscript but it'll also start enforcing no wildcard imports as well

**Comment by ByThePowerOfScience** [comment]
> > Currently, GTNH always ships with this mod installed, but it doesn't necessarily include JourneyMap
> 
> GTNH always ships with journeymap. no plans to change that either.

Of course, and that's fine. This just stops clients from crashing if they change it out for a different one but leave this mod installed without realizing.
That or you could make JourneyMap a required dependency of this one, but that might confuse players who didn't realize that JourneyMap was a hard requirement of the pack when it otherwise isn't.

**Comment by mitchej123** [comment]
Yah, it's the ic2 maven issue.  Run a `bash gradlew updateBuildscript` and it should work.  You might need to remove wildcard imports and replace them with the non wildcard version, but most IDEs can do that for you

File: src/main/kotlin/com/muxiu1997/sharewhereiam/util/WaypointUtil.kt [hunk 1]
```diff
@@ -15,6 +15,7 @@ import net.minecraft.util.ChatComponentText
 import net.minecraft.util.ChatStyle
 import net.minecraft.util.EnumChatFormatting
 import net.minecraft.util.IChatComponent
+import net.minecraftforge.fml.common.Loader
```
Comment (Dream-Master) [inline src/main/kotlin/com/muxiu1997/sharewhereiam/util/WaypointUtil.kt:18]: use import cpw.mods.fml.common.Loader instead

**Comment by ByThePowerOfScience** [comment]
Still can't load the environment, but you guys can edit my PR in my stead.

**Comment by Dream-Master** [comment]
@ByThePowerOfScience lets check if it build

**Review by mitchej123 — APPROVED** [review]

**Pull request merged**